build-scripts/build_rule_playbooks.py

```diff
@@ -0,0 +1,272 @@
+        values are dictionaries where keys are selectors and values are
+        variable values.
+        """
+        variables = dict()
+        for dirpath, dirnames, filenames in os.walk(self.guide_dir):
+            for filename in filenames:
+                root, ext = os.path.splitext(filename)
+                if ext == ".var":
+                    full_path = os.path.join(dirpath, filename)
+                    xccdf_value = ssg.build_yaml.Value.from_yaml(full_path)
+                    # Make sure that selectors and values are strings
+                    options = dict()
+                    for k, v in xccdf_value.options.items():
+                        options[str(k)] = str(v)
+                    variables[xccdf_value.id_] = options
+        return variables
+
+    def create_playbook_for_rule(self, snippet_path, rule_id, variables,
+                                 refinements, output_dir):
+        tags = self.get_tags_from_comments(snippet_path)
+
+        # TODO: Remove this temporary workaround
+        # TODO: Breaks tags and platforms !!!!!!
+        with open(snippet_path, "r") as snippet_file:
+            snippet_str = snippet_file.read()
+        snippet_str = snippet_str.replace("@ANSIBLE_TAGS@", "- dummy_tag")
+        snippet_str = snippet_str.replace("@ANSIBLE_ENSURE_PLATFORM@", "")
+        snippet_str = snippet_str.replace("@ANSIBLE_PLATFORM_CONDITION@",
+                                          "true")
+        snippet_str = snippet_str.replace("@RULE_ID@", "rule_id")
+        snippet_yaml = ssg.yaml.ordered_load(snippet_str)
+        # end of workaround
+
+        tasks, vars = self.get_data_from_snippet(snippet_yaml, variables,
+                                                 refinements)
+
+        if len(tasks) == 0:
+            raise ValueError(
+                "Ansible remediation for rule '%s' in '%s' "
+                "doesn't contain any task." %
+                (rule_id, snippet_path)
+            )
+
+        # play should be a dictionary, but to keep a logical order of elements
+        # we create the play as a list of tuples and later we convert it to
+        # an OrderedDict
+        play = [
+            ("name", rule_id),
+            ("hosts", "@@HOSTS@@"),
+            ("become", True),
+        ]
+        if len(vars) > 0:
+            play.append(("vars", vars))
+        if len(tags) > 0:
+            play.append(("tags", tags))
+        play.append(("tasks", tasks))
+        play_ordered_dict = OrderedDict(play)
+
+        playbook = [play_ordered_dict]
+        playbook_path = os.path.join(output_dir, rule_id + ".yml")
+        with open(playbook_path, "w") as playbook_file:
+            ssg.yaml.ordered_dump(
+                playbook, playbook_file, default_flow_style=False
+            )
+
+    def create_playbooks_for_profile(self, profile_path, variables):
+        """
+        Creates playbooks for profile from tasks from snippets.
```

Comment:
This is confusing.
Maybe you want to say `Creates a playbook for each rule selected in a profile`.